src/AzureBlobStorageUpdateSession/UpdateSession/UpdateSessionTable.cs

```diff
@@ -0,0 +1,123 @@
+        /// is to avoid situations where an instance fails to end its update session and blocks the cluster.</param>
+        public UpdateSessionTable(string connectionString, TimeSpan ttl)
+        {
+            _ttl = ttl;
+            CloudTableClient client = CloudStorageAccount.Parse(connectionString).CreateCloudTableClient();
+            _table = client.GetTableReference(UpdateSessionTableName);
+            _table.CreateIfNotExists();
+        }
+
+        public Task<UpdateSessionStatus> FetchUpdateSessionStatus(string clusterId, string appId)
+        {
+            string partitionQuery = CreatePartitionQuery(clusterId, appId);
+
+            TableQuery<UpdateDomainEntity> query = new TableQuery<UpdateDomainEntity>().Where(partitionQuery);
+            UpdateDomainEntity updateDomainEntity = null;
+            UpdateDomainEntity instanceListModifiedEntity = null;
+            List<UpdateDomainEntity> instancesList = new List<UpdateDomainEntity>();
+            foreach (UpdateDomainEntity entity in _table.ExecuteQuery(query))
+            {
+                if (entity.RowKey == UpdateDomainRowKey)
+                {
+                    updateDomainEntity = entity;
+                }
+                else if (entity.RowKey == ModifiedRowKey)
+                {
+                    instanceListModifiedEntity = entity;
+                }
+                else
+                {
+                    if (entity.Timestamp.Add(_ttl) > DateTimeOffset.Now)
+                    {
+                        instancesList.Add(entity);
+                    }
+                }
+            }
+
+            IEnumerable<UpdateDomainEntity> instancesEntities = instancesList;
+            if (updateDomainEntity != null)
+            {
+                // filter out entities that do not match the active update domain (were leftover for some reason)
+                instancesEntities = instancesEntities.Where(entity => entity.UpdateDomain == updateDomainEntity.UpdateDomain);
+            }
+
+            return Task.FromResult(new UpdateSessionStatus(instancesEntities, instanceListModifiedEntity, updateDomainEntity));
+        }
+
+        public async Task<bool> TryExecuteTransaction(UpdateSessionTransaction transaction)
+        {
+            try
+            {
+                await _table.ExecuteBatchAsync(transaction.BatchOperation);
+
+                return true;
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != 412 
+                    && e.RequestInformation.HttpStatusCode != 409)
+                {
+                    throw;
+                }
+            }
+            return false;
+        }
+
+        public async Task DeleteInstanceEntity(string clusterId, string instanceId, string appId)
+        {
+            TableOperation retrieveOperation = TableOperation.Retrieve<UpdateDomainEntity>(GetPartitionKey(clusterId, appId), 
```

Comment:
Maybe create a PartitionKey class initialized with clusterId and appId that sets the partition key once and exposes it through a getter? Then you pass/inject that instance around? GetPartitionKey is invariant in its result.